var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -755,6 +755,13 @@ def configure_args(self):
             "--disable-nls",
         ]
 
+        # Re-use the GNU target triplet defined in the existing gcc
+        existing_gcc = Executable("gcc")
+        triplet = existing_gcc("-###", error=str).partition("Target: ")[2].split("\n")[0]
+        if triplet:
+            options.append("--build={}".format(triplet))
+            options.append("--target={}".format(triplet))
```

Comment:
Should we wrap this in `if spec.satisfies("%gcc")` and use `self.compiler.cc` to be extra sure (since `gcc` might not actually be the compiler we build GCC with)?

Moreover, shouldn't we also set `--host` to the same value (see https://gcc.gnu.org/onlinedocs/gccint/Configure-Terms.html)?

Reply:
> Use the pre-existing gcc to define the correct build target.

Not sure if this is the way to go. Wouldn't this block us from having cross-compilers (which is something we want to do in the mid-term)? 

> This fixes https://github.com/spack/spack/issues/27668 as the Intel compiler needs the correct Debian tuplet x86_64-linux-gnu as target in its underlying gcc to find files under /usr/include/x86_64-linux-gnu.

This seems more an issue with Intel than with GCC. Can't we fix it in the Intel compiler class?